consensus/istanbul/announce/manager.go

```diff
@@ -479,7 +467,7 @@ func (m *Manager) StartAnnouncing(onStart func() error) error {
 	go m.announceThread()
 
 	if err := onStart(); err != nil {
-		m.StopAnnouncing(func() error { return nil })
```

Comment:
The reason for the split of StopAnnouncing and unlockedStopAnnouncing is to prevent a deadlock. If onStart returns an error, we need to stop everything, but StopAnnouncing requests a lock that we already have here. Therefore I separated the function from the one that requests the locks and an underlying one that has the business logic of stopping the announce

Reply:
It's reasonable. You could arguie it's a private function and you know who calls it, and those already check that (public function do checks, private one assume valid inputs). Or you could argue it's safer for the private function to do the check anyway.